common/src/main/java/io/pravega/common/hash/HashHelper.java

```diff
@@ -160,4 +167,77 @@ public int getAsInt() {
             return getHashCode().asInt();
         }
     }
+    
+    /**
+     * Hashes a bufferview in-place. (in contrast to murmur above which needs to copy data into a single array)
+     * (This passes the smhasher quality test suite)
+     *
+     * @param bufferView The input.
+     */
+    public static final long hashBufferView(BufferView bufferView) {
+        long multiple = 6364136223846793005L; // From knuth
```

Comment:
please pull all these magic numbers in constants.
Please add ample comments to what those numbers mean as well.

Also pointer to discussion of algorithm will be super useful.

Reply:
I have made these final constants at the top of the function. (I didn't promote them to the top of the file because they aren't used outside the function)
See the inner method's javadoc below.